ndbench-es-plugins/src/main/java/com/netflix/ndbench/plugin/es/EsAutoTuner.java

```diff
@@ -0,0 +1,78 @@
+package com.netflix.ndbench.plugin.es;
+
+import com.netflix.ndbench.api.plugin.NdBenchMonitor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+
+
+class EsAutoTuner {
+    private final ConstantStepWiseRateIncreaser rateIncreaser;
+    private final Float autoTuneFailureRatioThreshold;
+
+
+    private static final Logger logger = LoggerFactory.getLogger(EsAutoTuner.class);
+
+    /**
+     * // captures  time of first auto-tune recommendation request (made via a call to recommendNewRate);
+     * TODO - update comment
+     * Ignore the  possible race condition that arises if multiple threads call this method at around the same time
+     * when an instance is an uninitialized state. Two (or maybe more) instance of ConstantStepWiseRateIncreaser
+     * may be concurrently built, but the system clock time to rate mappings will be close enough that it
+     * won't really matter -- auto-tuning behavior will work as expected.
+     */
+    private volatile long timeOfFirstAutoTuneRequest = -1;
```

Comment:
Use AtomicLong instead?

Reply:
Kind of same thought process as I had for the atomic boolean flag. In fact.. i updated the comments to explicitly treat why I did not think any race condition that would occur is harmful.    Here is the new comment:    

    *
     * Note that we can ignore the  possible race condition that arises if multiple threads call this method at around
     * the same time.     In this case two threads will be attempting to set timeOfFirstAutoTuneRequest.. but the
     * target values they are using to set this variable  be so close it will not affect the desired behavior of the
     * auto-tuning feature.
     */
    double recommendNewRate(double currentRateLimit, WriteResult event, NdBenchMonitor runStats) {
        long currentTime = new Date().getTime();

        if (timeOfFirstAutoTuneRequest < 0) {          // race condition here when multiple writers, but can be ignored
            timeOfFirstAutoTuneRequest = currentTime;
        }



If you don't agree pls let me know why.. maybe i will learn something.    and of course i will change it if you wish.. even if i don't end up seeing things the same way.